tests/common/ixia/common_helpers.py

```diff
@@ -72,6 +72,147 @@ def get_vlan_subnet(host_ans):
     gw_addr = ansible_stdout_to_str(mg_vlan_intfs[0]['addr'])
     return gw_addr + '/' + str(prefix_len)
 
+def get_mac(host_ans):
+    """
+    Get the MAC address of the device
+
+    Args:
+        host_ans: Ansible host instance of the device
+
+    Returns:
+        MAC address (str), e.g., "11:22:33:44:55:66", or None
+    """
+    config_facts = host_ans.config_facts(host=host_ans.hostname,
+                                         source="running")['ansible_facts']
+
+    if 'DEVICE_METADATA' not in config_facts:
+        return None
+
+    device_metadata = config_facts['DEVICE_METADATA']
+    if 'localhost' not in device_metadata:
+        return None
+
+    localhost_metadata = device_metadata['localhost']
+    if 'mac' not in localhost_metadata:
+        return None
+
+    return str(localhost_metadata['mac'])
+
+def get_logical_intf_member(host_ans, member_type):
+    """
+    Get logical interfaces (vlan/portchannel) and their attached physical
+    interfaces (a.k.a., members)
+
+    Args:
+        host_ans: Ansible host instance of the device
+
+    Returns:
+        A dictionary where key is the logical interface and value is the
+        list of physical interfaces attached to this logical interface
+
+        For example:
+        {
+            'PortChannel0001': ['Ethernet0', 'Ethernet4'],
+            'PortChannel0002': ['Ethernet8', 'Ethernet12']
+        }
+        {
+            'Vlanl0001': ['Ethernet0', 'Ethernet4'],
+            'Vlanl0002': ['Ethernet8', 'Ethernet12']
+        }
+    """
+    config_facts = host_ans.config_facts(host=host_ans.hostname,
+                                         source="running")['ansible_facts']
+
+    if member_type not in config_facts:
+        return None
+
+    member = config_facts[member_type]
+    result = {}
+
+    for logical_intf in member:
+        physical_intfs = list(member[logical_intf].keys())
+        result[logical_intf] = physical_intfs
+
+    return result
+
+def get_intf_ipv4_addr(host_ans):
```

Comment:
not needed. This is also in mg_facts as stated above

Sample dump:
"minigraph_portchannel_interfaces": [{"addr": "10.0.0.56", "attachto": "PortChannel0001", "mask": "255.255.255.254", "peer_addr": "10.0.0.57", "prefixlen": 31, "subnet": "10.0.0.56/31"}
"minigraph_vlan_interfaces": [{"addr": "192.168.0.1", "attachto": "Vlan1000", "mask": "255.255.248.0", "prefixlen": 21, "subnet": "192.168.0.0/21"}],